lib/netext/dialer.go

```diff
@@ -165,6 +177,11 @@ func (d *Dialer) findRemote(addr string) (*lib.HostAddress, error) {
 
 	remote, err := d.getConfiguredHost(addr, host, port)
 	if err != nil || remote != nil {
+		if err == nil && d.BlockedHostnames != nil && net.ParseIP(host) == nil {
```

Comment:
`net.ParseIP(host)` is already called on line 188. Couldn't you move that before the `d.getConfiguredHost()` call and then you'll know you're dealing with a hostname here?

Reply:
fixed in ce17b17a